### DISCUSSION TITLE
Restrict select access to user

### DISCUSSION THREAD

**Discussion by bkniffler** [supabase/supabase#3933] [Questions]
While querying GET:/connections?select=* with a self signed JWT:

```
DROP ROLE some_role;
CREATE ROLE some_role;
GRANT some_role TO supabase_admin;

ALTER TABLE connections ENABLE ROW LEVEL SECURITY;
CREATE POLICY "connections_teamId_policy" ON connections USING ("teamId" = auth.uid());

GRANT SELECT ON connections TO some_role;
REVOKE SELECT (credentials) ON connections FROM some_role;
```

This doesn't restrict any access to column `credentials`, it still shows in result set.

```
DROP ROLE some_role;
CREATE ROLE some_role;
GRANT some_role TO supabase_admin;

ALTER TABLE connections ENABLE ROW LEVEL SECURITY;
CREATE POLICY "connections_teamId_policy" ON connections USING ("teamId" = auth.uid());

GRANT SELECT (id, name) ON connections TO some_role;
```

This yields `permission denied for table connections`.

What's the right way to grant access to some columns to a user and/or restrict access to some other columns?

**Comment by keybits** [comment]
There might be a more elegant solution, but [this blog post](https://www.benburwell.com/posts/row-level-security-postgresql-views/) describes a way to achieve what you want using Functions and Views.

**Comment by steve-chavez** [comment]
Check this previous discussion on column-level security: https://github.com/supabase/supabase/discussions/3158#discussioncomment-1311387. 

> GRANT some_role TO supabase_admin;

Why is this needed?
